### PR TITLE
feat: add docs to use react native signer

### PR DESCRIPTION
**Pull request by iykazrji**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [x] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [x] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [x] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [x] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR adds documentation for using the `React Native Signer` package, including setup, user authentication methods, and handling deep links.

### Detailed summary
- Updated `react-native-signer.ts` to include a link to the new documentation.
- Created `using-the-signer.mdx` with:
  - Title and description for the guide.
  - Instructions for creating a `Signer` instance.
  - Details on user authentication via email and deep links.
  - Example code snippets for authentication and deep link handling.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by iykazrji** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1182?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1182** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1182?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1182?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#1181** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1181?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1180** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1180?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `moldy/rn-base`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by moldy530 — COMMENTED** [review]
we should also add a Next Steps section in the bare install and expo install that links to this doc for getting started

**Comment by iykazrji** [comment]
@moldy530 Created a new PR to add the Getting started section to the Bare and Expo documentation [here](https://github.com/alchemyplatform/aa-sdk/pull/1190)

File: site/pages/react-native/signer/using-the-signer.mdx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  client: { connection: { apiKey: API_KEY } },
+});
+```
+
+## Authenticate a User
+
+Next, you'll need to authenticate your user before you can use the Signer as an owner on the account
+
+:::info
+At the time of writing, the React Native Signer only supports Email at the moment.
```
Comment (moldy530) [inline site/pages/react-native/signer/using-the-signer.mdx:25]:
```suggestion
At the time of writing, the React Native Signer only supports Email magic link.
```

**Review by moldy530 — APPROVED** [review]
only had one nit

**Comment by iykazrji** [comment]
Closing this in favor of: https://github.com/alchemyplatform/aa-sdk/pull/1181

**Pull request closed without merging**